common/djangoapps/config_models/models.py

```diff
@@ -176,3 +179,58 @@ def key_values(cls, *key_fields, **kwargs):
         values = list(cls.objects.values_list(*key_fields, flat=flat).order_by().distinct())
         cache.set(cache_key, values, cls.cache_timeout)
         return values
+
+    def fields_equal(self, instance, fields_to_ignore=("id", "change_date", "changed_by")):
+        """
+        Compares this instance's fields to the supplied instance to test for equality.
+        This will ignore any fields in `fields_to_ignore`.
+
+        Note that this method ignores many-to-many fields.
+
+        Args:
+            instance: the model instance to compare
+            fields_to_ignore: List of fields that should not be compared for equality. By default
+            includes `id`, `change_date`, and `changed_by`.
+
+        Returns: True if the checked fields are all equivalent, else False
+        """
+        for field in self._meta.get_fields():
+            if not field.many_to_many and field.name not in fields_to_ignore:
+                if getattr(instance, field.name) != getattr(self, field.name):
+                    return False
+
+        return True
+
+    @classmethod
+    def equal_to_current(cls, json, fields_to_ignore=("id", "change_date", "changed_by")):
+        """
+        Compares for equality this instance to a model instance constructed from the supplied JSON.
+        This will ignore any fields in `fields_to_ignore`.
+
+        Note that this method cannot handle fields with many-to-many associations, as those can only
+        be set on a saved model instance (and saving the model instance will create a new entry).
+        All many-to-many field entries will be removed before for the equality comparison is done.
+
+        Args:
+            json: json representing an entry to compare
+            fields_to_ignore: List of fields that should not be compared for equality. By default
+            includes `id`, `change_date`, and `changed_by`.
+
+        Returns: True if the checked fields are all equivalent, else False
+        """
+
+        # Remove many-to-many relationships from json.
+        # They require an instance to be already saved.
+        info = model_meta.get_field_info(cls)
+        for field_name, relation_info in info.relations.items():
+            if relation_info.to_many and (field_name in json):
+                json.pop(field_name)
+
+        new_instance = cls(**json)
```

Comment:
It seems a bit odd to create a model instance here. Could `equal_to_current` just compare the fields on `cls.current(*key_field_args)` with the items in `json`? Or does the class constructor do data conversion on them?